src/main/java/com/amazon/dlic/auth/http/saml/HTTPSamlAuthenticator.java

```diff
@@ -200,6 +202,9 @@ public Optional<SecurityResponse> reRequestAuthentication(final SecurityRequest
                 new SecurityResponse(HttpStatus.SC_UNAUTHORIZED, Map.of("WWW-Authenticate", getWwwAuthenticateHeader(saml2Settings)), "")
             );
         } catch (Exception e) {
```

Comment:
This is done this way because `authTokenProcessorHandler.handle` throws a general exception: https://github.com/opensearch-project/security/blob/418af25c2a4d4774faedddf1b565f2895b69f067/src/main/java/com/amazon/dlic/auth/http/saml/HTTPSamlAuthenticator.java#L193

For the case of `SecurityRequestChannelUnsupported`, the header verifier has logic to case this to handle gracefully and it should continue to be thrown.